src/Pad.js

```diff
@@ -4,22 +4,60 @@ import './Pad.css';
 class Pad extends Component {
     constructor(props) {
         super(props);
+
         this.state = {
-            clicked: false
+            clicked: false,
+        };
+
+        this.padClicked = this.padClicked.bind(this);
+    }
+
+    componentDidMount() {
+        document.addEventListener('keydown', this.keyTrigger.bind(this), false);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('keydown', this.keyTrigger.bind(this), false);
+    }
+
+    keyTrigger(e) {
+        if (e.key === this.props.trigger) {
+            this.padClicked();
         }
     }
 
     padClicked() {
-        this.setState({clicked : true});
-        setTimeout(()=>{
-            this.setState({clicked : false});
-        }, 100);
+        const { context } = this.props;
+
+        this.setState({ clicked: true });
+        setTimeout(() => {
+            this.setState({ clicked: false });
+        }, 150);
+
+        const source = context.createBufferSource();
+
+        // Prepare a fetch if the file has not been cached, otherwise this will give 300 level fetch
+        fetch(this.props.file).then((response) => {
```

Comment:
Something I'm thinking of trying is calling for each of these samples in the componentWillMount for each ```<Pad />```